tests/test_rules.py

```diff
@@ -672,6 +672,25 @@ def test_references(self):
         self.assertEqual(faulty_rules, [], Fore.RED +
                          "There are rules with malformed 'references' fields. (has to be a list of values even if it contains only a single value)")
 
+    def test_references_in_description(self):
+        # This test checks for the presence of a links and special keywords in the "description" field while there is no "references" field.
+        faulty_rules = []
+        for file in self.yield_next_rule_file_path(self.path_to_rules):
+            references = self.get_rule_part(
+                file_path=file, part_name="references")
+            # Reference field doesn't exist
+            if not references:
+                descriptionfield = self.get_rule_part(
+                    file_path=file, part_name="description")
+                if descriptionfield:
+                    for i in ["http://", "https://", "internal research"]: # Extends the list with other common references starters
+                        if i in descriptionfield.lower():
+                            print(Fore.RED + "Rule {} has malformed description field that contain references to external links.".format(file))
+                            faulty_rules.append(file)
+
+        self.assertEqual(faulty_rules, [], Fore.RED +
+                         "There are rules with malformed 'description' fields. (links and external references have to be in a seperate field named 'references'. see specification https://github.com/SigmaHQ/sigma-specification)")
```

Comment:
Is there a deep link to the part you are mentioning? e.g. https://github.com/SigmaHQ/sigma-specification/blob/main/Sigma_1_0_2.md#references-optional

Reply:
And will have to be updated once the new version/branch schema is implemented

Reply:
Didn't want to put a permalink since we're gonna update the structure. I just put the full repo and once the new versioning is implemented, I'll add a fixed link to the "references"

Reply:
ok. I would do it the other way round but it doesn't matter if it is correct in the end.